DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -265,59 +265,39 @@ class TrackerTopology {
-    } else { return 0; }
+    return ( ((id.rawId() >>tobVals_.sterStartBit_ ) & tobVals_.sterMask_ ) == 1 ) ? 1 : 0;
   }
 
   uint32_t tibStereo(const DetId &id) const {
-    if ( ((id.rawId() >>tibVals_.sterStartBit_ ) & tibVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ );
-    } else { return 0; }
+    return ( ((id.rawId() >>tibVals_.sterStartBit_ ) & tibVals_.sterMask_ ) == 1 ) ? 1 : 0;
   }
 
   uint32_t tidStereo(const DetId &id) const {
-    if ( ((id.rawId() >>tidVals_.sterStartBit_ ) & tidVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_ );
-    } else { return 0; }
+    return ( ((id.rawId() >>tidVals_.sterStartBit_ ) & tidVals_.sterMask_ ) == 1 ) ? 1 : 0;
   }
 
   uint32_t tecStereo(const DetId &id) const {
-    if ( ((id.rawId() >>tecVals_.sterStartBit_ ) & tecVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ );
-    } else { return 0; }
+    return ( ((id.rawId() >>tecVals_.sterStartBit_ ) & tecVals_.sterMask_ ) == 1 ) ? 1 : 0;
   }
 
   uint32_t tibGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ ) == 1 ) {
-      return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ ) == 2 ) {
-      return ( id.rawId() - 2 );
-    } else { return 0; }
+    uint32_t testId = (id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_;
+    return ( testId == 1 || testId == 2) ? (id.rawId() - testId) : 0;
   }
 
   uint32_t tecGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ ) == 1 ) {
-      return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ ) == 2 ) {
-      return ( id.rawId() - 2 );
-    } else { return 0; }
+    uint32_t testId = (id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_;
+    return ( testId == 1 || testId == 2) ? (id.rawId() - testId) : 0;
   }
 
   uint32_t tobGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_ ) == 1 ) {
-      return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_ ) == 2 ) {
-      return ( id.rawId() - 2 );
-    } else { return 0; }
+    uint32_t testId = (id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_;
+    return ( testId == 1 || testId == 2) ? (id.rawId() - testId) : 0;
   }
 
   uint32_t tidGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_ ) == 1 ) {
-      return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_ ) == 2 ) {
-      return ( id.rawId() - 2 );
-    } else { return 0; }
+    uint32_t testId = (id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_;
+    return ( testId == 1 || testId == 2) ? ( id.rawId() - testId ) : 0;
```

Comment:
@VinInn your last comment was "It can only be 0, 1 or 2 .."
do you mean the output of <code> tidGlued </code> or the result of the calculation <code> (id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_ </code>

if it is the calculation, then it is more efficient to compare testId with 0.

Thanks